src/key/helper.js

```diff
@@ -368,25 +368,30 @@ export function sanitizeKeyOptions(options, subkeyDefaults = {}) {
   return options;
 }
 
-export function isValidSigningKeyPacket(keyPacket, signature) {
+export function isValidSigningKeyPacket(keyPacket, signature, config) {
   if (!signature.verified || signature.revoked !== false) { // Sanity check
     throw new Error('Signature not verified');
   }
-  return keyPacket.algorithm !== enums.read(enums.publicKey, enums.publicKey.rsaEncrypt) &&
-    keyPacket.algorithm !== enums.read(enums.publicKey, enums.publicKey.elgamal) &&
-    keyPacket.algorithm !== enums.read(enums.publicKey, enums.publicKey.ecdh) &&
+  const keyAlgo = enums.write(enums.publicKey, keyPacket.algorithm);
+  return checkKeyStrength(keyPacket, config) &&
```

Comment:
Mmm perhaps it's indeed best to leave the function as-is, and do the key strength check separately, since it's a different domain (key capabilities vs custom requirements).

Reply:
Yeah that's also fine :+1:
But I think refactoring these functions is also not a bad idea, that way you could get: `Error encrypting message: Could not find valid encryption key packet in key {keyid}: DSA is not a valid encryption algorithm` or something like that